### DISCUSSION TITLE
Material UI Select reset doesn't fill the select value

### DISCUSSION THREAD

**Discussion by thodwris** [react-hook-form/react-hook-form#3898] [Q&A]
Hi,

I want to reset the form while I'm changing the select but the value is not being rendered.
I think this is happening because it triggers rerender and this code is getting called:
```
useEffect(() => {
    register({ name: "select" });
}, [register]);
```

I've tried using Memo but it didn't do the trick unless I used it in a wrong way.

Example demonstrating the issue:
[https://codesandbox.io/s/react-hook-form-controlled-mixed-with-uncontrolled-forked-kr30v?file=/src/index.js](https://codesandbox.io/s/react-hook-form-controlled-mixed-with-uncontrolled-forked-kr30v?file=/src/index.js)

I'd appreciate any help.

Thank you

**Comment by bluebill1049** [comment]
you will need to use Controller, custom register is like a virtual input, it doesn't update any value.

**Comment by thodwris** [comment]
Can you give me an example?

**Reply by bluebill1049** [reply]
https://react-hook-form.com/api#Controller

**Reply by thodwris** [reply]
Ok I used `Controller` and I can now get the value rendered on the select menu but `reset()` doesn't seem working.
I fill the input field and then I change the select menu but it doesn't clear the input field.
Updated code sandbox [https://codesandbox.io/s/react-hook-form-controlled-mixed-with-uncontrolled-forked-kr30v](https://codesandbox.io/s/react-hook-form-controlled-mixed-with-uncontrolled-forked-kr30v)

**Reply by bluebill1049** [reply]
did you see the example in the doc? https://codesandbox.io/s/react-hook-form-v6-controller-qsd8r

**Comment by thodwris** [comment]
Yes I did but I don't see any solution to my problem there 🤔

I remind you my issue is that reset doesn't work after change the mui
select value.

Btw thank you so much for your time.

**Reply by DenisaPr** [reply]
Hi, I had the same issue, the solution that worked for me is to use the value from Controller and pass it on to MUI Select.
```javascript
<Controller control={control} name={field.name} rules={{ ...validations }}
    render={({ value, onChange }) => (
        <Select
            label={label}
            onChange={e => onChange(e.target.value)}
           value={value}>
               {options}
        </Select>
    )}
 />
```

**Reply by bluebill1049** [reply]
@thodwris I think your select problem missing one of the props, thanks @DenisaPr for the solution.

**Reply by abhilashsajeev** [reply]
The problem is How do reset this value. I ve tried using this code but on reset function call the UI is not getting updated.


for Example the following code will show a warning of setting uncontrolled component to controlled for mui Select. While the native Select component works seemlessly

```js
import "./styles.css";
import React, {useEffect, useState} from "react"
import * as Yup from "yup";
import {Select, FormControl, InputLabel, MenuItem} from "@material-ui/core"
import { yupResolver } from "@hookform/resolvers/yup";
import { Checkbox as MuiCheckbox, FormControlLabel } from "@material-ui/core";
import {useForm, Controller} from "react-hook-form"
export default function App() {
  const validationSchema = Yup.object().shape({
    income: Yup.string().required('Income Pleaded is required'),
   income_type: Yup.string(),
   state_id: Yup.number()

});
const {
    register,
     reset, watch,control,
    handleSubmit,
    formState: { errors }
} = useForm({
    resolver: yupResolver(validationSchema)
});

useEffect(()=>{
  setTimeout(()=>{
    reset({income: 'new val', state_id:'32',test:true,
    income_type:'32'})
  }, 1000)
})
const onSubmit = (data)=>{
  console.log("dadasdf", data)
}
  return (
    <div className="App">
      <h1>Hello CodeSandbox</h1>
      <input {...register('income')} label="text" type="text" />
      <br/>
      <select {...register('income_type')} defaultValue="">
        <option value=""></option>
        <option value="32">Kerala</option>
        <option value="33">TN</option>
      </select>
     
      <br/>
         
                
      <RHFSelect label="reset me please"  name="state_id" control={control} />
      <button onClick={handleSubmit(onSubmit)}>submit</button>
         </div>
  );
}


export const RHFSelect = ({
  name,
  label,
  control,
  
}) => {
  return (
    <Controller control={control} name={name} 
    render={({ value, onChange }) => (
        <Select
            label={label}
            
           value={value}>
              <MenuItem value="32">Kerala</MenuItem>
              <MenuItem value="35">KR</MenuItem>
              <MenuItem value="35">TN</MenuItem>
              <MenuItem value="36">AP</MenuItem>

       </Select>
          )}
        
      
    />
  );
};

```
